### PR TITLE
Credential for adding Private Repo

### PR DESCRIPTION
**Pull request by matheussn**
### Description

Now the command `rit add repo` has a specific credential, so that the user does not need to regenerate his private tokens each time the command is executed.

### How to verify it
In its first execution, when adding a private repository, the command will request the token, as it already did. However, the command now saves this token in a credential.
After executing the `rit add repo` command adding a new private repository, use the `rit list credential` command, to verify that a new credential has been created in the current environment.

### Changelog

Save and reuse repo credentials for private repos

### CODE REVIEW

File: pkg/cmd/add_repo.go [hunk 1]
```diff
@@ -142,7 +143,7 @@ func (ad addRepoCmd) runPrompt() CommandRunnerFunc {
 
 		var token string
 		if isPrivate {
-			token, err = ad.Password("Personal access tokens:")
+			token, err = ad.Resolve("CREDENTIAL_" + provider + "-add-repo_token")
```
Comment (matheussn) [inline pkg/cmd/add_repo.go:146]: I didn't really like this implementation, but I'm afraid to try a more "beautiful" implementation and end up getting more complex than necessary.

I accept suggestions.

**Comment by henriquemoraeszup** [comment]
Hey, we had a recent merge that affects add repo, can you update it?

File: pkg/cmd/add_repo.go (hunk 1 above)
Comment (henriquemoraeszup) [reply pkg/cmd/add_repo.go:146]: I think reusing the resolver is the way to go, if anything changes on the logic, we still have one source of truth.

Comment (henriquemoraeszup) [reply pkg/cmd/add_repo.go:146]: On another note, it is convention to use credential constants always capitalized, like `CREDENTIAL_GITHUB_ADD_REPO_TOKEN`

File: pkg/cmd/add_repo.go [hunk 2]
```diff
@@ -153,6 +154,13 @@ func (ad addRepoCmd) runPrompt() CommandRunnerFunc {
 		gitRepoInfo := git.NewRepoInfo(url, token)
 		tags, err := git.Repos.Tags(gitRepoInfo)
 		if err != nil {
+			if strings.Contains(err.Error(), "401") {
+				errorString := fmt.Sprintf("permission error:\nYou must overwrite the current token (%s-add-repo) with command:\n"+
+					"\t rit set credential\n"+
+					"Or move to a new environment with command:\n"+
```
Comment (henriquemoraeszup) [inline pkg/cmd/add_repo.go:160]: Nitpick:
```suggestion
					"Or switch to a new environment with the command:\n"+
```

File: pkg/credential/settings.go [hunk 3]
```diff
@@ -215,13 +215,15 @@ func NewDefaultCredentials() Fields {
...
-		"ansible":    []Field{username, password},
+		AddNew:            []Field{},
+		"github":          []Field{username, email, token},
+		"gitlab":          []Field{username, email, token},
+		"aws":             []Field{accessKey, secretAccessKey},
+		"jenkins":         []Field{username, token},
+		"kubeconfig":      []Field{base64config},
+		"ansible":         []Field{username, password},
+		"github-add-repo": []Field{token},
+		"gitlab-add-repo": []Field{token},
```
Comment (henriquemoraeszup) [inline pkg/credential/settings.go:226]: This is a "reserved" credential, not sure we should expose to the user, what do you think? Can we edit credentials in Ritchie after it has been set?

Comment (matheussn) [reply pkg/credential/settings.go]: I think it makes sense to keep them "reserved"

**Comment by brunasilvazup** [comment]
> ### Description
> Now the command `rit add repo` has a specific credential, so that the user does not need to regenerate his private tokens each time the command is executed.
> 
> ### How to verify it
> In its first execution, when adding a private repository, the command will request the token, as it already did. However, the command now saves this token in a credential.
> After executing the `rit add repo` command adding a new private repository, use the `rit list credential` command, to verify that a new credential has been created in the current environment.
> 
> ### Changelog
> Save and reuse repo credentials for private repos

Hi @matheussn, could you add gifs with the simulation of your change operation please?

File: pkg/cmd/add_repo.go [hunk 4]
```diff
@@ -235,6 +236,13 @@ func (ar *addRepoCmd) resolvePrompt() (formula.Repo, error) {
 	gitRepoInfo := git.NewRepoInfo(url, token)
 	tags, err := git.Repos.Tags(gitRepoInfo)
 	if err != nil {
+		if strings.Contains(err.Error(), "401") {
+			errorString := fmt.Sprintf("permission error:\nYou must overwrite the current token (%s-add-repo) with command:\n"+
```
Comment (kaduartur) [inline pkg/cmd/add_repo.go:240]: could you extract `errorString` to const and create it using the raw string literals **``**?

File: pkg/cmd/add_repo.go [hunk 5]
```diff
@@ -235,6 +236,13 @@ func (ar *addRepoCmd) resolvePrompt() (formula.Repo, error) {
 	gitRepoInfo := git.NewRepoInfo(url, token)
 	tags, err := git.Repos.Tags(gitRepoInfo)
 	if err != nil {
+		if strings.Contains(err.Error(), "401") {
```
Comment (kaduartur) [inline pkg/cmd/add_repo.go:239]: I think you can create a specific error for 401 Unauthorized and check if err is equal

**Review by kaduartur — CHANGES_REQUESTED** [review]

**Comment by lucasdittrichzup** [comment]
what's up @matheussn? hope you're well! if you have any questions to adjust the PR according to the team's suggestions, just get in touch with us, okay?

**Comment by GuillaumeFalourd** [comment]
Giving continuity to the feature implementation here: https://github.com/ZupIT/ritchie-cli/pull/986

**Pull request closed without merging**